fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/metadata/SerializableFunction.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * SerializableFunction.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2025 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.metadata;
+
+import com.apple.foundationdb.record.PlanSerializationContext;
+import com.apple.foundationdb.record.RecordMetaDataProto;
+import com.apple.foundationdb.record.query.plan.cascades.MacroFunction;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+public interface SerializableFunction {
```

Comment:
Can this be called something that encompasses more than just serialization. I had suggested `CatalogedFunction`.

Reply:
discussed offline, renamed.